### PR TITLE
osd: Fixes for osd_scrub_during_recovery handling

### PR DESCRIPTION
**Pull request by dzafman**
Fixes: http://tracker.ceph.com/issues/18206

### CODE REVIEW

**Comment by dzafman** [comment]
retest this please

File: qa/standalone/scrub/osd-recovery-scrub.sh [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+    setup $dir || return 1
+    run_mon $dir a --osd_pool_default_size=1 || return 1
+    run_mgr $dir x || return 1
+    for osd in $(seq 0 $(expr $OSDS - 1))
+    do
+        run_osd $dir $osd || return 1
+    done
+
+    # Create a pool with a single pg
```
Comment (gregsfortytwo) [inline qa/standalone/scrub/osd-recovery-scrub.sh]: Sure looks to me like this is creating a pool with 32 PGs?

**Comment by dzafman** [comment]
@tchaikov I made the change that you recommended in 2 places.  Just need to squash.

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2017-08-20_09:42:12-rados-wip-kefu-testing-distro-basic-mira/

**Pull request merged**